be/src/storage/rowset/bitmap_index_writer.cpp

```diff
@@ -76,29 +126,24 @@ class BitmapIndexWriterImpl : public BitmapIndexWriter {
     void add_values(const void* values, size_t count) override {
         auto p = reinterpret_cast<const CppType*>(values);
         for (size_t i = 0; i < count; ++i) {
-            add_value(unaligned_load<CppType>(p));
+            const CppType& value = unaligned_load<CppType>(p);
+            auto it = _mem_index.find(value);
```

Comment:
1. If the cardinality is high, does the find() dominate? It's better to test the performance at different cardinality levels.
2. We have the performance test file, you can use it and also integrate the performance test into the framework.

Reply:
I have tested the performance under different cardinality levels.
1. With hardcode roaring array
https://github.com/StarRocks/starrocks/pull/5329#issuecomment-1107834777
2. After removing hardcode roaring array, without replacing std::map with phmap::flat_hash_map
https://github.com/StarRocks/starrocks/pull/5329#issuecomment-1108001054

When the cardinality is high, the find search does take a bit of time. And I try to replace std::map with phmap::flat_hash_map. In experiment2 and experiment3, it has small gains(improve only 1 or 2 seconds). However, in exeriment4(the bitmap index is a random int(range from (-2147483648, 2147483647)), the version with the phmap::flat_hash_map optimization has much lower performance(around 460s, while the unoptimized version is 267s, the std::map version is 254s). In the profile, I found that phmap::flat_hash_map does frequent growing size, and this part has a lot of overhead when there is high cardinality.  So i finally still choose the std::map.